python/pyspark/sql/context.py

```diff
@@ -104,12 +128,12 @@ def _ssql_ctx(self):
         return self._jsqlContext
 
     @property
-    def _conf(self):
+    def _conf(self):  # type: ignore[no-untyped-def]
         """Accessor for the JVM SQL-specific configurations"""
         return self.sparkSession._jsparkSession.sessionState().conf()
 
     @classmethod
-    def getOrCreate(cls, sc):
+    def getOrCreate(cls: type, sc: SparkContext) -> SQLContext:
```

Comment:
I don't think we have to annotate `cls` (same way, as won't annotate `self`, with a few exceptions). However, if we're going to do that, we should probably use specific type (i.e. `"Type[Foo]"`)